packages/react-native/Libraries/AppDelegate/RCTAppDelegate.mm

```diff
@@ -117,6 +118,7 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     }
     rootView = [self createRootViewWithBridge:self.bridge moduleName:self.moduleName initProps:initProps];
   }
+  [self customizeRootView:(RCTRootView *)rootView];
```

Comment:
nit: Can't we set the rootView variable type to `RCTRootView`? And cast it to UIView if needed

Reply:
Nope, this won't change a lot.

The variable is defined as `UIView *rootView` because it is the common ancestor.

The `surfaceHostingProxyRootView;` is actually a `@interface RCTSurfaceHostingProxyRootView : RCTSurfaceHostingView` where `RCTSurfaceHostingView` is a `@interface RCTSurfaceHostingView : UIView <RCTSurfaceDelegate>`. 
So, technically, it is not even a `RCTRootView`, but a proxied one, and it will still require casting.

The `createRootViewWithBridge:moduleName:initProps:` technically returns a `UIView *`, so changing if we made the `rootView` a `RCTRootView *` we would have either to change the `createRootViewWithBridge` (breaking change, we don't want to do it) or add the casting to `RCTRootView`.

So, in the end, we will always have to cast. :(

Reply:
Okay, thanks for the explanation